spikeinterface/toolkit/postprocessing/spike_locations.py

```diff
@@ -0,0 +1,180 @@
+    extension_name = 'spike_locations'
+    
+    def __init__(self, waveform_extractor):
+        BaseWaveformExtractorExtension.__init__(self, waveform_extractor)
+
+        self._locations = None
+        self._all_spikes = None
+
+    def _set_params(self, ms_before=1, ms_after=1, method='center_of_mass',
+                    method_kwargs={}):
+
+        params = dict(ms_before=ms_before,
+                      ms_after=ms_after,
+                      method=method,
+                      method_kwargs=method_kwargs)
+        return params        
+        
+    def _specific_load_from_folder(self):
+        recording = self.waveform_extractor.recording
+        sorting = self.waveform_extractor.sorting
+
+        all_spikes = sorting.get_all_spike_trains(outputs='unit_index')
+        self._all_spikes = all_spikes
+
+        self._locations = []
+        for segment_index in range(recording.get_num_segments()):
+            file_locs = self.extension_folder / f'locations_segment_{segment_index}.npy'
+            locs_seg = np.load(file_locs)
+            self._locations.append(locs_seg)
+
+    def _reset(self):
+        self._locations = None
+    
+    def _specific_select_units(self, unit_ids, new_waveforms_folder):
+        # load filter and save amplitude files
+        for seg_index in range(self.waveform_extractor.recording.get_num_segments()):
+            loc_file_name = f"locations_segment_{seg_index}.npy"
+            locs = np.load(self.extension_folder / loc_file_name)
+            _, all_labels = self.waveform_extractor.sorting.get_all_spike_trains()[seg_index]
+            filtered_idxs = np.in1d(all_labels, np.array(unit_ids)).nonzero()
+            np.save(new_waveforms_folder / self.extension_name /
+                    loc_file_name, locs[filtered_idxs])
+    
+        
+    def compute_locations(self, **job_kwargs):
+        """
+        This function first transforms the sorting object into a `peaks` numpy array and then
+        uses the`sortingcomponents.peak_localization.localize_peaks()` function to triangulate
+        spike locations.
+        """
+        from spikeinterface.sortingcomponents.peak_localization import localize_peaks
+        
+        we = self.waveform_extractor
+        recording = we.recording
+        sorting = we.sorting
+
+        all_spikes = sorting.get_all_spike_trains(outputs='unit_index')
+        self._all_spikes = all_spikes
+        
+        extremum_channel_inds = get_template_extremum_channel(we, outputs="index")
+        sorting_peaks = we.sorting.to_spike_vector()
+        unit_inds = sorting_peaks['unit_ind']
+        channel_inds = [extremum_channel_inds[u_id] for u_id in we.sorting.unit_ids[unit_inds]] 
+        sorting_peaks['unit_ind'] = channel_inds
+        dtype_names = list(sorting_peaks.dtype.names)
+        unit_ind_index = dtype_names.index("unit_ind")
+        dtype_names[unit_ind_index] = "channel_ind"
+        sorting_peaks.dtype.names = tuple(dtype_names)
```

Comment:
I would move this to `Sorting.to_spike_vector()` with an option : `with_channel_ind=False/True`.
In that case I would not replace 'unit_ind' but add clearly 'channel_ind'

I will amke a proposal separatly for this, if you are ok with this.